pkgs/development/interpreters/python/cpython/3.6/default.nix

```diff
@@ -64,6 +64,14 @@ in stdenv.mkDerivation {
     substituteInPlace configure --replace '-Wl,-stack_size,1000000' ' '
   '';
 
+  patches = optionals (!(stdenv.cc.isGNU or false)) [
```

Comment:
How I understand it is that it currently only works because gcc doesn't complain when compiling c++ code using `$CC`, clang does which is arguably the correct thing to do 😄 .
In cases like this I prefer not to use conditionals, making everything more consistent and the darwin build less brittle. (if the patch fails to apply the linux build won't break)